File: app/src/main/java/com/example/proyecto/domain/Util/New/Pruebas/QuoteService.java
package com.example.proyecto.domain.Util.New.Pruebas;

import com.example.proyecto.data.model.Unit;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;

public class QuoteService {

    private final Retrofit retrofit = RetrofitHelper.getRetrofit();

    public ArrayList<Unit> getQuotes(){
        ArrayList<Unit> units = new ArrayList<>();
//        Response<ArrayList<Unit>> response = RetrofitHelper.getRetrofit().create(Api.class).getAllQuotes();
        Call<ArrayList<Unit>> response = retrofit.create(Api2.class).getAllQuotes();
        ArrayList<Unit> finalUnits = units;
        response.enqueue(new Callback<ArrayList<Unit>>() {
            @Override
            public void onResponse(Call<ArrayList<Unit>> call, Response<ArrayList<Unit>> responsee) {
                for (int i = 0; i>responsee.body().size(); i++){
                    finalUnits.add(responsee.body().get(i));
                }
            }

            @Override
            public void onFailure(Call<ArrayList<Unit>> call, Throwable t) {

            }
        });

//        units = response.body();
        if (units == null) units = null;

        return units;
    }
}

File: app/src/main/java/com/example/proyecto/data/network/QuoteService.kt
package com.example.proyecto.data.network

import com.example.proyecto.core.RetrofitHelper
import com.example.proyecto.data.model.Model_Base64
import com.example.proyecto.data.model.QuoteModelUnidadSesion
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

class QuoteService {

    private val retrofit = RetrofitHelper.getRetrofit()
//    private val retrofit_api = RetrofitHelper.getRetrofit_API()

    suspend fun getQuotes(codigo:String):List<QuoteModelUnidadSesion> {
        val COD=codigo
        return withContext(Dispatchers.IO){
            val response = retrofit.create(Api::class.java).getAllQuotes("listarSesion.php?Cod="+codigo)
            response.body() ?: emptyList()
        }
    }
//    suspend fun getPrediccion(model_Base64: Model_Base64):String {
//        val COD=model_Base64
//        return withContext(Dispatchers.IO){
//            val response_api = retrofit_api.create(Api::class.java).getPrediccion(model_Base64)
//            response_api.body() ?: "error en la respuesta"
//        }
//    }
}